File: system-test/sec-tests/python/sec_22_idor.py
# SEC-22: IDOR — Insecure Direct Object Reference
# Test: TC-CR-029
from flask import Flask, request
from models import User

app = Flask(__name__)

@app.route('/api/user/<user_id>')
def get_user(user_id):
    # No ownership check — any user can access any user_id
    return User.query.get(user_id)

File: system-test/sec-tests/javascript/sec_22_idor.js
// SEC-22: IDOR — Insecure Direct Object Reference
// Test: TC-CR-029
const express = require('express');
const app = express();

app.get('/api/user/:userId', (req, res) => {
  const userId = req.params.userId;
  // No ownership check — any user can access any userId
  const user = db.collection('users').findById(userId);
  res.json(user);
});
